libraries/standard/http/src/http_client.c

```diff
@@ -16,22 +45,158 @@ HTTPStatus_t HTTPClient_AddHeader( HTTPRequestHeaders_t * pRequestHeaders,
     return HTTP_NOT_SUPPORTED;
 }
 
+/*-----------------------------------------------------------*/
+
 HTTPStatus_t HTTPClient_AddRangeHeader( HTTPRequestHeaders_t * pRequestHeaders,
                                         int32_t rangeStart,
                                         int32_t rangeEnd )
 {
     return HTTP_NOT_SUPPORTED;
 }
 
+/*-----------------------------------------------------------*/
+
+static HTTPStatus_t _sendHttpHeaders( const HTTPTransportInterface_t * pTransport,
+                                      const HTTPRequestHeaders_t * pRequestHeaders )
+{
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+    int32_t transportStatus = 0;
+
+    /* Send the HTTP headers over the network. */
+    transportStatus = pTransport->send( pTransport->pContext,
+                                        pRequestHeaders->pBuffer,
+                                        pRequestHeaders->headersLen );
+
+    if( transportStatus < 0 )
+    {
+        IotLogErrorWithArgs( "Error in sending the HTTP headers over the transport "
+                             "interface. Transport status %d.",
+                             transportStatus );
+        returnStatus = HTTP_NETWORK_ERROR;
+    }
+    else if( transportStatus != pRequestHeaders->headersLen )
+    {
+        IotLogErrorWithArgs( "Attempted to send %d of HTTP headers, transport "
```

Comment:
Please keep the data section separate from the message (as mentioned in the standard). Suggest changing to:
"Failure in sending HTTP headers: Transport layer sent lesser than required bytes: AttemptedBytes=%d, SentBytes=%d"

Reply:
Similar suggestion for the `_sendHttpBody` function

Reply:
Good luck enforcing this